pkg/client/typed/discovery/discovery_client.go

```diff
@@ -212,6 +212,9 @@ func setDiscoveryDefaults(config *restclient.Config) error {
 	config.APIPath = ""
 	config.GroupVersion = nil
 	config.Codec = runtime.NoopEncoder{api.Codecs.UniversalDecoder()}
+	if config.UserAgent == "" {
```

Comment:
As per Russ cox, it does not matter it seems: https://groups.google.com/forum/#!topic/golang-nuts/7Ks1iq2s7FA

```
The one that makes the code clear.
If I'm about to look at element x I typically write
len(s) > x, even for x == 0, but if I care about
"is it this specific string" I tend to write s == "".

It's reasonable to assume that a mature compiler will compile
len(s) == 0 and s == "" into the same, efficient code.
Right now 6g etc do compile s == "" into a function call
while len(s) == 0 is not, but that's been on my to-do list to fix.

Make the code clear.
```